laravel/app/Http/Controllers/ScheduleDataController.php

```diff
@@ -581,270 +433,89 @@ private function insertOption()
+                        ->count() != 0
+                ) {
+                    $notify = true;
+                    array_push($overwrittenChoices, $i + 1);
+
+                    Option::where("date", $date)
+                        ->where("resident", $resident)
+                        ->where("option", $choice)
+                        ->update([
+                            "schedule" => $split[$i],
+                            "attending" => $attending,
+                            "milestones" => $_REQUEST["milestones" . $choice],
+                            "objectives" => $_REQUEST["objectives" . $choice],
+                            "anesthesiologist_id" => $pref_anest[$i],
+                            "isValid" => 1,
+                        ]);
+                } else {
+                    // Insert data
+                    if (!is_null($schedule_data[$i])) {
+                        Option::insert([
+                            "date" => $date,
+                            "resident" => $resident,
+                            "schedule" => $split[$i],
+                            "attending" => $attending,
+                            "option" => $choice,
+                            "milestones" => $_REQUEST["milestones" . $choice],
+                            "objectives" => $_REQUEST["objectives" . $choice],
+                            "anesthesiologist_id" => $pref_anest[$i],
+                            "isValid" => 1,
+                        ]);
+                    }
+                }
             } else {
                 Option::where("date", $date)
                     ->where("resident", $resident)
                     ->where("option", $choice)
                     ->delete();
-                // ->update(['isValid'=>0]);
-            }
-        } else {
-            $overwrittenChoices[2] = 0;
-            if (!is_null($schedule_data3)) {
-                // Insert data
-                Option::insert([
-                    "date" => $date,
-                    "resident" => $resident,
-                    "schedule" => $split[2],
-                    "attending" => $attending,
-                    "option" => $choice,
-                    "milestones" => $_REQUEST["milestones3"],
-                    "objectives" => $_REQUEST["objectives3"],
-                    "anesthesiologist_id" => $pref_anest3,
-                    "isValid" => 1,
-                ]);
             }
         }
 
-        // data was overwritten, send a notification
+        // if data was overwritten, send a notification
         if ($notify == true) {
-            // please make sure to change the email here
             self::notifyResidentOverwrittenPreferences(
-                "",
+                $residentName,
                 $_SERVER["HTTP_EMAIL"],
                 $residentName,
                 $date,
                 $overwrittenChoices
             );
         }
-    }
 
-    public function clearOption($date)
```

Comment:
I'm not certain that it's safe to remove this entire method. It seems the `'schedule/secondday/preferences/clear/{date}'` route uses it.

Reply:
I'll refactor and put it back, but I don't think the route that calls it is exists.